plenum/server/request_managers/write_request_manager.py

```diff
@@ -302,44 +308,12 @@ def do_taa_validation(self, request: Request, req_pp_time: int, config):
                 )
                 return
 
-        taa = None
-        taa_data = self.get_taa_data()
-        if taa_data is not None:
-            (taa, taa_seq_no, taa_txn_time), taa_digest = taa_data
-
-        if taa is None:
-            if request.taaAcceptance:
-                raise InvalidClientTaaAcceptanceError(
-                    request.identifier, request.reqId,
-                    "Txn Author Agreement acceptance has not been set yet"
-                    " and not allowed in requests"
-                )
-            else:
-                logger.trace(
-                    "{} TAA acceptance passed for request {}: taa is not set"
-                    .format(self, request.reqId)
-                )
-                return
-
-        if not taa[TXN_AUTHOR_AGREEMENT_TEXT]:
-            if request.taaAcceptance:
-                raise InvalidClientTaaAcceptanceError(
-                    request.identifier, request.reqId,
-                    "Txn Author Agreement acceptance is disabled"
-                    " and not allowed in requests"
-                )
-            else:
-                logger.trace(
-                    "{} TAA acceptance passed for request {}: taa is disabled"
-                    .format(self, request.reqId)
-                )
-                return
-
-        if not taa_digest:
-            raise LogicError(
-                "Txn Author Agreement digest is not defined: version {}, seq_no {}, txn_time {}"
-                .format(taa[TXN_AUTHOR_AGREEMENT_VERSION], taa_seq_no, taa_txn_time)
+        if not self.get_taa_data():
```

Comment:
I think we need to get uncommitted TAA here, as committed is not guaranteed to be the same on all the nodes. So, if there is a request with accepted TAA sent right after this TAA is added to the ledger, there is a chance that it may be handled differently by nodes.